experimental/dds/tree2/src/shared-tree-core/sharedTreeCore.ts

```diff
@@ -414,7 +425,18 @@ export class SharedTreeCore<TEditor extends ChangeFamilyEditor, TChange> extends
 			const [newHead] = rebaser.rebaseBranch(branch.getHead(), this.getLocalBranchHead());
 			const changes: GraphCommit<TChange>[] = [];
 			findAncestor([newHead, changes], (c) => c === this.getLocalBranchHead());
-			this.applyPathFromBranch(branch, changes);
+
+			this.undoRedoManager.updateAfterRebase(
+				this.getLocalBranchHead(),
+				newHead,
+				this.undoRedoManager,
+				branch.undoRedoManager,
+			);
+
+			// Apply the changes without tracking them in the undo redo manager
```

Comment:
Why?

Reply:
I updated the comment here to explain why. It's just because the call to `updateAfterRebase` already handles the tracking in a specialized way.

Reply:
Thanks - that sounds great, it just took me a bit to figure out why when I first saw it so I figured more elaboration will be helpful for others in the future.